src/main/java/aztech/modern_industrialization/machines/init/SingleBlockCraftingMachines.java

```diff
@@ -179,11 +193,14 @@ public static void registerMachineTiers(String englishName, String machine, Mach
             MachineGuiParameters builtGuiParams = guiParamsBuilder.build();
             String id = prefix + "_" + machine;
 
+            var overclockComponent = (extraConfig.steamOverclockComponent != null) ? extraConfig.steamOverclockComponent
```

Comment:
This feels wrong - the config object should be properly constructed before it's passed to this function. (See below)